server/src/main/java/com/objectcomputing/checkins/services/volunteering/VolunteeringServiceImpl.java

```diff
@@ -0,0 +1,193 @@
+        this.rolePermissionServices = rolePermissionServices;
+        this.organizationRepo = organizationRepo;
+        this.relationshipRepo = relationshipRepo;
+        this.eventRepo = eventRepo;
+    }
+
+    @Override
+    public List<VolunteeringOrganization> listOrganizations(boolean includeDeactivated) {
+        return organizationRepo.findAll(includeDeactivated);
+    }
+
+    @Override
+    public List<VolunteeringRelationship> listRelationships(UUID memberId, UUID organizationId, boolean includeDeactivated) {
+        if (memberId != null && organizationId != null) {
+            return relationshipRepo.findByMemberIdAndOrganizationId(memberId, organizationId, includeDeactivated);
+        } else if (memberId != null) {
+            return relationshipRepo.findByMemberId(memberId, includeDeactivated);
+        } else if (organizationId != null) {
+            return relationshipRepo.findByOrganizationId(organizationId, includeDeactivated);
+        } else {
+            return relationshipRepo.findAll(includeDeactivated);
+        }
+    }
+
+    @Override
+    public List<VolunteeringEvent> listEvents(@Nullable UUID memberId, @Nullable UUID organizationId, boolean includeDeactivated) {
+        if (memberId != null && organizationId != null) {
+            return eventRepo.findByMemberIdAndOrganizationId(memberId, organizationId, includeDeactivated);
+        } else if (memberId != null) {
+            return eventRepo.findByMemberId(memberId, includeDeactivated);
+        } else if (organizationId != null) {
+            return eventRepo.findByOrganizationId(organizationId, includeDeactivated);
+        } else {
+            return eventRepo.findAll(includeDeactivated);
+        }
+    }
+
+    @Override
+    public VolunteeringOrganization create(VolunteeringOrganization organization) {
+        if (organization.getId() != null) {
+            return update(organization);
+        }
+        // Fail if a certification with the same name already exists
+        validate(organizationRepo.getByName(organization.getName()).isPresent(),
+                ORG_NAME_ALREADY_EXISTS_MESSAGE,
+                organization.getName());
+        return organizationRepo.save(organization);
+    }
+
+    @Override
+    public VolunteeringRelationship create(VolunteeringRelationship relationship) {
+        if (relationship.getId() != null) {
+            return update(relationship);
+        }
+        validateRelationship(relationship, "create");
+        return relationshipRepo.save(relationship);
+    }
+
+    @Override
+    public VolunteeringEvent create(VolunteeringEvent event) {
+        if (event.getId() != null) {
+            return update(event);
+        }
+        validateEvent(event, "create");
+        return eventRepo.save(event);
+    }
+
+    @Override
+    public VolunteeringOrganization update(VolunteeringOrganization organization) {
+        // Fail if a certification with the same name already exists (but it's not this one)
```

Comment:
Small but here and in the L79 comment, entity changed from `certification` to a volunteering org.

Reply:
Good eyes! Fixed in https://github.com/objectcomputing/check-ins/pull/2490/commits/680d54b73f55a71b1486409e966bdc40944cd020